### DISCUSSION TITLE
[Icon Request] Classic floppy disk save icon

### DISCUSSION THREAD

**Discussion by icabetong** [tailwindlabs/heroicons#476] [Ideas]
A classic floppy disk save icon is very useful for forms. 😁

**Comment by aldomendez** [comment]
I created one by self... 
![image](https://user-images.githubusercontent.com/204436/96896511-350fc880-1453-11eb-8c1f-be9c9ecf4459.png)
for some reason doesn't look quite right...

**Comment by icabetong** [comment]
A little curves on a edges perhaps?

**Comment by ChristopheVanassche** [comment]
The current save-icon looks a bit misleading indeed, would also like to see a classic version of the save-icon.

**Comment by optimizasean** [comment]
The current save icon, I feel it is actually a download file icon.  Since so many people associate the floppy classic as save since it has been and still is being used in many programs and places, I think this should be added in for good UI/UX design since most users will recognize it (still keep the existing icon though!).

Also nice floppy, I think the edge on the right is too long?  Something does look strange, just not sure what 😄

**Reply by howdu** [reply]
Still with v2 there's no save icon.... but they must agree the old save looked more like a download icon now that has been removed.

**Comment by skstuder** [comment]
+1, love the library, I have a use case that requires a floppy disk.

**Comment by Quared** [comment]
Cmon. Really no save icon?

**Comment by ghost** [comment]
Yesterday I built these three versions trying to follow the style of the set. 

![image](https://user-images.githubusercontent.com/927078/224253233-b22c5b07-d2c6-402f-a41c-08fef6bdeccf.png)

Can I create a pull request for review?

**Reply by skstuder** [reply]
> Yesterday I built these three versions trying to follow the style of the set.
> 
> ![image](https://user-images.githubusercontent.com/927078/224253233-b22c5b07-d2c6-402f-a41c-08fef6bdeccf.png)
> 
> Can I create a pull request for review?

Maintainers do not seem to be very responsive here, I'd say just make that PR and we will all comment on it for visibility. Hope it gets through 🤞

**Reply by optimizasean** [reply]
Been watching this in the background, I will definitely upvote any PR for a floppy. Yours look pretty good.

Floppy photo: https://imageio.forbes.com/specials-images/imageserve/5f3545995dc16a259697d3c2/Floppy-disk/960x0.jpg?format=jpg&width=960

The *shutter* isn't wide enough (because the cut is too deep) makes it look slightly off, also cut is supposed to be on the right if you look at the front where the label is (front).
Otherwise it is on the left if you are looking at the back where the *hub* is.

Edit: shutter, not data access cover. Hub not tape turning doohicky lol. We use the standard 3.5in floppy as save icon usually for reference (most prevalent and final version of the floppy to be successful).
Thanks wikipedia, never thought I would be describing the anatomy of a floppy decades later: https://en.wikipedia.org/wiki/Floppy_disk

**Reply by pmgower** [reply]
> Yesterday I built these three versions trying to follow the style of the set.
> 
> ...
> 
> Can I create a pull request for review?

@jacorachan did you create your pull request?

If not, please create it and share it here!

I want to comment on it or up-vote it to get the maintainers' attention so we can get them to merge it! :)

**Comment by sayhicoelho** [comment]
Any updates? Still not having a floppy disk icon 🙁

**Comment by ZhenjaMax** [comment]
+1!
It's weird that HeroIcons haven't this really useful and necessary icon. :(

**Comment by boonkerz** [comment]
any updates?

**Comment by skstuder** [comment]
It's been months. No one that maintains this is paying attention. Best to
just look elsewhere for this icon.

On Wed, Jul 19, 2023 at 10:02 AM boonkerz ***@***.***> wrote:

> any updates?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tailwindlabs/heroicons/discussions/476#discussioncomment-6491163>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANH6J5GCUEULNNOUPM7AJ73XRAHLNANCNFSM6AAAAAATT5OC4U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

-- 


********************************************



 

*Inmar Confidentiality 
Note*:  This e-mail and any attachments are confidential and intended to be 
viewed and used solely by the intended recipient.  If you are not the 
intended recipient, be aware that any disclosure, dissemination, 
distribution, copying or use of this e-mail or any attachment is 
prohibited.  If you received this e-mail in error, please notify us 
immediately by returning it to the sender and delete this copy and all 
attachments from your system and destroy any printed copies.  Thank you for 
your cooperation.



 

*Notice of Protected Rights*:  The removal of any 
copyright, trademark, or proprietary legend contained in this e-mail or any 
attachment is prohibited without the express, written permission of Inmar, 
Inc.  Furthermore, the intended recipient must maintain all copyright 
notices, trademarks, and proprietary legends within this e-mail and any 
attachments in their original form and location if the e-mail or any 
attachments are reproduced, printed or distributed.

 

********************************************

**Reply by sayhicoelho** [reply]
Do you know a good alternative?

**Reply by optimizasean** [reply]
not as clean as heroicons but here is Icons8 which is probably closest but also font awesome (you can just copy the svg instead of the whole set and <i> notation they use). Flaticon/Flat-icons is nice as well. There is also feather icons.

Other alternatives: https://alternativeto.net/software/heroicons/

**Reply by rekkih** [reply]
Lucide Icons seems to be the best bet, at least for me: https://github.com/lucide-icons/lucide. Entirely community made, too.

**Reply by sayhicoelho** [reply]
Wow! Awesome! It's a good replacement! Thanks <3

**Comment by skstuder** [comment]
Just used a circle with a checkmark for my use case, not great but gets the
job done.
[image: Screenshot 2023-07-19 at 2.21.11 PM.png]

On Wed, Jul 19, 2023 at 12:19 PM Renan Coelho ***@***.***>
wrote:

> Do you know a good alternative?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tailwindlabs/heroicons/discussions/476#discussioncomment-6492208>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANH6J5DJ4F5YQGZIDWKZDI3XRAXMZANCNFSM6AAAAAATT5OC4U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

-- 


********************************************



 

*Inmar Confidentiality 
Note*:  This e-mail and any attachments are confidential and intended to be 
viewed and used solely by the intended recipient.  If you are not the 
intended recipient, be aware that any disclosure, dissemination, 
distribution, copying or use of this e-mail or any attachment is 
prohibited.  If you received this e-mail in error, please notify us 
immediately by returning it to the sender and delete this copy and all 
attachments from your system and destroy any printed copies.  Thank you for 
your cooperation.



 

*Notice of Protected Rights*:  The removal of any 
copyright, trademark, or proprietary legend contained in this e-mail or any 
attachment is prohibited without the express, written permission of Inmar, 
Inc.  Furthermore, the intended recipient must maintain all copyright 
notices, trademarks, and proprietary legends within this e-mail and any 
attachments in their original form and location if the e-mail or any 
attachments are reproduced, printed or distributed.

 

********************************************

**Reply by ben519** [reply]
Alternatively, `DocumentCheckIcon` is decent.

**Comment by jammastergirish** [comment]
I was surprised not to see the standard Save icon too!

**Reply by sayhicoelho** [reply]
Try Lucide Icons https://lucide.dev/icons/

**Comment by collimarco** [comment]
Any updates on this? It's really strange that there isn't a save icon

**Comment by amirkouretchian** [comment]
boo to no save icons

**Comment by jamiemlaw** [comment]
Here are some alternatives I put together, that you are free to use:

![Screenshot 2024-03-05 at 16 02 27](https://github.com/tailwindlabs/heroicons/assets/956081/fa7d5d32-18d6-4d51-bfa7-1641ed7932db)

```xml
<!-- outline -->
<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="h-6 w-6">
  <path stroke-linecap="round" stroke-linejoin="round" d="M3 3v18h18V6l-3-3H3ZM7.5 3v6h9V3M6 21v-9h12v9M14.25 5.25v1.5"/>
</svg>

<!-- solid -->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="h-6 w-6">
  <path d="M18 2.25c.2 0 .39.08.53.22l3 3c.14.14.22.33.22.53v15c0 .41-.34.75-.75.75H3a.75.75 0 0 1-.75-.75V3c0-.41.34-.75.75-.75h15ZM17 12H7a1 1 0 0 0-1 1v7.25h12V13a1 1 0 0 0-1-1Zm-.5-8.25h-9V8a1 1 0 0 0 1 1h7a1 1 0 0 0 1-1V3.75Zm-2.25.75c.41 0 .75.34.75.75v1.5a.75.75 0 1 1-1.5 0v-1.5c0-.41.34-.75.75-.75Z" />
</svg>

<!-- mini -->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="h-5 w-5">
  <path d="M14.59 2a1 1 0 0 1 .7.3l2.42 2.4a1 1 0 0 1 .29.71V17a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1ZM14 10H6a1 1 0 0 0-1 1v5.5h10V11a1 1 0 0 0-1-1Zm0-6.5H6V7a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V3.5Zm-2 .75c.41 0 .75.34.75.75v1.25a.75.75 0 1 1-1.5 0V5c0-.41.34-.75.75-.75Z" />
</svg>

<!-- micro -->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="currentColor" class="h-4 w-4">
  <path fill="#000" fill-rule="evenodd" d="M12.59 1a1 1 0 0 1 .7.3l1.42 1.4a1 1 0 0 1 .29.71V14a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1ZM11.5 8h-7a1 1 0 0 0-1 1v4.5h9V9a1 1 0 0 0-1-1Zm.5-5.5H4v3a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-3Zm-2 .75c.41 0 .75.34.75.75v1a.75.75 0 1 1-1.5 0V4c0-.41.34-.75.75-.75Z" />
</svg>
```

**Comment by daveycodez** [comment]
Is this project abandoned?

**Reply by ben519** [reply]
There's a floppy disk save icon now.

https://lucide.dev/icons/save

## EDIT

I am now realizing I wrote this comment on the repo for heroicons when I thought it was the repo for lucide icons. Derp. For people reading this, I highly recommend you look into lucide icons as it's very similar but with more icons and more active development.

**Comment by optimizasean** [comment]
@icabetong 

> There's a floppy disk save icon now.
> 
> https://lucide.dev/icons/save

They added the icon!!! 🎉🎉🎉 Can close issue now?

**Reply by kaustubhxd** [reply]
It's still not added in Heroicons. Issue remains open no?

**Comment by leoriviera** [comment]
Hello! Yes, uh, a floppy disk icon would be pretty nice!

**Comment by Foofding** [comment]
I also need a floppy disk icon to use for a save button.

**Comment by TeaDrinkingProgrammer** [comment]
+1

**Comment by dylviz** [comment]
I ended up here too :)

**Comment by daveycodez** [comment]
I ended up just using the checkmark.. lol

**Comment by paulhuisman** [comment]
Just responding here because we need a Floppy icon.

**Comment by jakefeeley** [comment]
Is there any chance of this icon appearing in the library anytime soon?

**Comment by HalmarBergman** [comment]
If I were to provide an icon set, a classic and recognisable 'Save' icon would be among the most important and firsts added. So yeah, the 'Floppy' icon would be greatly appreciated

**Comment by raphtastisch** [comment]
+1
And out of curiosity: I would be really interested into the reason why this icon was not already in the very first version - its so obvious to have this, thus there must be a reason the maintainers actively decided against it.

**Comment by morceaudebois** [comment]
### We yearn for the floppy 💾

**Comment by kaustubhxd** [comment]
If it's a one-off use case, we can just use the emoji 💾 and change the size using font size

**Comment by ndri** [comment]
I added `floppy-disk` to [Sidekickicons](https://github.com/ndri/sidekickicons/), which are meant to be used as a companion to Heroicons. Browse them here: https://sidekickicons.com/?icon=floppy-disk

**Reply by daveycodez** [reply]
These are sick, if you're open to suggestions..

Teardrop, Skull, Tree, Video Reel, Audio Wave, Ear, Bunny

Would be dope to have

**Reply by olinoles** [reply]
@ndri cool package! keep working on it please 🙏

**Comment by yogesh-dhande** [comment]
"bookmark-square" icon is close enough for me
